sympy/polys/matrices/sdm.py

```diff
@@ -1243,6 +1268,22 @@ def sdm_transpose(M):
     return MT
 
 
+def sdm_dotvec(A, B, K):
+    total = K.zero
+    for j in A.keys() & B.keys():
+        total += A[j] * B[j]
+    return total
```

Comment:
```suggestion
    return sum(A[j]*B[j] for j in A.keys() & B.keys(), start=K.zero)
```

Reply:
Yes, good point. Actually `K.sum` is better (it uses `Add(*args)` for `EXRAW`).